File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcGroup/IfcSystem/IfcBuildingSystem/IfcBuildingSystem.java
public class IfcBuildingSystem extends IfcSystem {
    public IfcBuildingSystemTypeEnum PredefinedType;
    public IfcLabel LongName;

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcBuildingSystemTypeEnum.USERDEFINED) OR ((PredefinedType = IfcBuildingSystemTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
}

File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcGroup/IfcSystem/IfcBuildingSystem/IfcBuildingSystem.js
class IfcBuildingSystem extends IfcSystem {
    constructor() {
        /** @type {IFCBUILDINGSYSTEMTYPEENUM} */
        this.PredefinedType = null;
        /** @type {IFCLABEL} */
        this.LongName = null;
    }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcBuildingSystemTypeEnum.USERDEFINED) OR ((PredefinedType = IfcBuildingSystemTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
}
